src/fhirtypes/common.ts

```diff
@@ -218,6 +218,28 @@ export function setPropertyOnInstance(
         } else {
           if (typeof assignedValue === 'object') {
             Object.assign(current[key][index], assignedValue);
+            // check if we have used modifier extensions correctly
+            if ((key === 'extension' || key === 'modifierExtension') && assignedValue.url != null) {
+              const extension = fisher.fishForFHIR(assignedValue.url, Type.Extension);
+              if (extension) {
+                const isModifierExtension =
+                  extension.snapshot.element.find((el: ElementDefinition) => el.id === 'Extension')
+                    .isModifier === true;
+                if (key === 'extension' && isModifierExtension) {
+                  throw new Error(
+                    `Instance of modifier extension ${
+                      extension.name ?? extension.id
+                    } assigned to extension path. Modifier extensions should only be assigned to modifierExtension paths.`
+                  );
+                } else if (key === 'modifierExtension' && !isModifierExtension) {
+                  throw new Error(
```

Comment:
Should we make these errors just for consistency? I'm not sure that anywhere else we throw errors without defining our own error class to throw.